nexus_constructor/add_component_window.py

```diff
@@ -122,6 +126,28 @@ def setupUi(self, parent_dialog):
         # Validate the default values set by the UI
         self.unitsLineEdit.validator().validate(self.unitsLineEdit.text(), 0)
         self.nameLineEdit.validator().validate(self.nameLineEdit.text(), 0)
+        self.addFieldPushButton.clicked.connect(self.add_field)
+        self.removeFieldPushButton.clicked.connect(self.remove_field)
+
+        # Set whatever the default nx_class is so the fields autocompleter can use the possible fields in the nx_class
+        self.on_nx_class_changed()
+
+        self.fieldsListWidget.itemClicked.connect(self.select_field)
```

Comment:
I think that if you implement the comment below to remove `self.selected_widget` this is no longer required.

Reply:
Removing this disables selection completely. This doesn't have anything to do with `self.selected_widget`